### PR TITLE
Font map casing

### LINKED ISSUE #7376: `FontMap.getFont` is case sensitive
**Describe the bug**
`FontMap.getFont` is case sensitive. If you have an iModel with the font "Arial", calling `FontMap.getFont(“arial”)` does not find the font.

**To Reproduce**

1. Go to [My iTwins](https://developer.bentley.com/my-itwins/).
2. Create an iTwin and create an iModel from the existing House Sample:
Image
3. This iModel has a font with the name “Arial”. Open the iModel in any app of your choice.
4. Call load the font map via a call to `await IModelConnection.fontMap.loadFontMap()`. Then call `IModelConnection.fontMap.getFont("arial")`. Observe that the font is not found.

OR

1. Run the modified test in the associated PR.

**Expected behavior**
`FontMap.getFont` should find a font by name regardless of casing.

**Additional context**
I have pushed a PR with the fix. I modified the tests to catch this case. You can reproduce this issue by running the modified test.

### PR DESCRIPTION
**Pull request by Alfonso-Martello**
Fixes #7376

### CODE REVIEW

File: common/changes/@itwin/core-backend/fontMapCase_2024-11-14-21-53.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/core-backend",
+      "comment": "IModel.test.ts now validates that FontMap.getFont uses case insensitive matching for font names",
```
Comment (aruniverse) [inline common/changes/@itwin/core-backend/fontMapCase_2024-11-14-21-53.json:5]:
```suggestion
      "comment": "",
```

change logs are for end consumers, and they wont care about tests like this

**Review by aruniverse — APPROVED** [review]

File: common/changes/@itwin/core-backend/fontMapCase_2024-11-14-21-53.json (hunk 1 above)
Comment (Alfonso-Martello) [reply common/changes/@itwin/core-backend/fontMapCase_2024-11-14-21-53.json]: Thanks, suggestion committed

**Review by pmconne — CHANGES_REQUESTED** [review]
This is a known issue, see discussion in #6751. Please don't merge yet. JavaScript's case-insensitive comparison uses a different collation sequence than that used by iModels.

In what contexts is this causing a problem? Are people manually typing in font names?

**Comment by Alfonso-Martello** [comment]
It is only causing problems in a niche situation for now. No one is manually typing font names. In drawing production, we are using `TextStyleSettings` stored in workspaces. `TextStyleSettings` reference their font by name.

Somehow we found two iModels that reference the same font but have different spellings (the first iModel has the font named “Arial” embedded while the second iModel has the font named “arial” embedded).

Our UI only shows text styles that have their font available in the iModel, so we were wondering why our “Arial” text style was not being listed for the iModel with “arial”. 

We saw that line of code and thought it was a quick fix so opened this PR/issue. Happy to close as the issue is not pressing.

**Comment by pmconne** [comment]
@khanaffan @ColinKerr what was the outcome of your iQacs collection re https://github.com/iTwin/imodel-native/pull/722? Do we have any plans to change the collation sequence for newly-created iModels?

**Comment by pmconne** [comment]
The [native API](https://github.com/iTwin/imodel-native/blob/eae3dd7c30e8da58892f823f701cc5712ddfd406/iModelCore/iModelPlatform/DgnCore/FontManager.cpp#L152) for embedding fonts does a Unicode case-insensitive comparison to reject the input font if a font already exists with the same name. So you could only end up with an issue if two briefcases imported the same font with different casing (where the difference is in something other than ASCII characters), then one pushes their changes and the other pulls them. But that would only occur if both briefcases neglect to first obtain the lock on the entire iModel before writing to the props table. Since no public TypeScript APIs exist for embedding fonts, we're talking about connectors, which surely must obtain that lock.

So I'm fine with merging this change. @khanaffan do you disagree?

**Comment by Alfonso-Martello** [comment]
So, I have some thoughts/questions.
1. I know there was a font discussion the other day with @Josh-Schifter  and @williamkbentley . This is second hand, but my understanding was that a TypeScript API for embedding fonts would be added (or maybe another way for embedding fonts would be exposed)? If that occurs, then would this be more of an issue (i.e. not just applicable to the connectors)?
2. I do not see a clear way to guarantee consistent behavior between JavaScript's collation and our iModel collation. Which then makes me wonder if it is possible for `FontMap.getFont` to return accurate information in all scenarios (either we have inconsistent collation or we have potentially missed fonts due to casing). Maybe it does not matter if the only way to get into a bad data situation is to have a niche workflow as outlined above.
3. I see in #6751 that the plan is to deprecate `FontMap`. I'm not sure if this is still the plan after the most recent discussion, but if yes, then this PR is unneeded.

**Comment by khanaffan** [comment]
> @khanaffan @ColinKerr what was the outcome of your iQacs collection re [iTwin/imodel-native#722](https://github.com/iTwin/imodel-native/pull/722)? Do we have any plans to change the collation sequence for newly-created iModels?

We found several imodel with Latin1 case sensitive codeValues. But it could be done for new imodels.

**Review by pmconne — APPROVED** [review]

**Comment by khanaffan** [comment]
> The [native API](https://github.com/iTwin/imodel-native/blob/eae3dd7c30e8da58892f823f701cc5712ddfd406/iModelCore/iModelPlatform/DgnCore/FontManager.cpp#L152) for embedding fonts does a Unicode case-insensitive comparison to reject the input font if a font already exists with the same name. So you could only end up with an issue if two briefcases imported the same font with different casing (where the difference is in something other than ASCII characters), then one pushes their changes and the other pulls them. But that would only occur if both briefcases neglect to first obtain the lock on the entire iModel before writing to the props table. Since no public TypeScript APIs exist for embedding fonts, we're talking about connectors, which surely must obtain that lock.
> 
> So I'm fine with merging this change. @khanaffan do you disagree?

EmbedFont() seem to also use MAX(id) value for given namespace / name to generate new id that it use for new font been stored. This id is part of primary key. I mean without lock two briefcases can easily cause primary key conflict. The strData (contain json) property change from two briefcase only cause data conflict. We can add rule to handle data conflict by skipping/replace. But for Primary key we can only skip or abort (abort been default).

BeProp is not really good place to store fonts. Workspaces may be better and they are by default only let one writer.

**Comment by pmconne** [comment]
> EmbedFont() seem to also use MAX(id) value for given namespace / name to generate new id that it use for new font been stored. This id is part of primary key. I mean without lock two briefcases can easily cause primary key conflict.

We're supposed to use schema sync db to prevent conflicts between server-issued Ids.

> BeProp is not really good place to store fonts. Workspaces may be better and they are by default only let one writer.

Workspaces involve uploading resources to blob storage and redistributing them from there, something expressly prohibited by virtually every licensed font. For example, [per Microsoft](https://learn.microsoft.com/en-us/typography/fonts/font-faq#redistribution-and-extended-rights) most fonts delivered with Windows permit redistribution strictly through embedding.

**Pull request merged**